pkg_resources is deprecated for importlib

https://importlib-resources.readthedocs.io/en/latest/migration.html

```
/usr/local/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1006: in _gcd_import
    ???
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:677: in _load_unlocked
    ???
/usr/local/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
    exec(co, module.__dict__)
src/ewoksxrpd/tests/conftest.py:6: in <module>
    import pyFAI.azimuthalIntegrator
/usr/local/lib/python3.7/site-packages/pyFAI/__init__.py:47: in <module>
    from .utils import decorators
/usr/local/lib/python3.7/site-packages/pyFAI/utils/__init__.py:55: in <module>
    from .. import resources
/usr/local/lib/python3.7/site-packages/pyFAI/resources/__init__.py:72: in <module>
    import pkg_resources
/usr/local/lib/python3.7/site-packages/pkg_resources/__init__.py:122: in <module>
    stacklevel=2
E   DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
```

--- a/ci/requirements_appveyor.txt
+++ b/ci/requirements_appveyor.txt
@@ -13,3 +13,4 @@ matplotlib
 numexpr != 2.8.6
 silx >= 2
 psutil
+importlib_resources

--- a/ci/requirements_appveyor.txt
+++ b/ci/requirements_appveyor.txt
@@ -13,4 +13,4 @@ matplotlib
 numexpr != 2.8.6
 silx >= 2
 psutil
-importlib_resources
+importlib_resources; python_version < '3.9'

--- a/ci/requirements_gh.txt
+++ b/ci/requirements_gh.txt
@@ -20,4 +20,4 @@ matplotlib
 numexpr != 2.8.6
 silx >= 2
 psutil
-importlib_resources
+importlib_resources; python_version < '3.9'

--- a/ci/requirements_gh.txt
+++ b/ci/requirements_gh.txt
@@ -20,3 +20,4 @@ matplotlib
 numexpr != 2.8.6
 silx >= 2
 psutil
+importlib_resources

--- a/ci/requirements_rtd.txt
+++ b/ci/requirements_rtd.txt
@@ -11,4 +11,4 @@ nbsphinx
 sphinx-rtd-theme
 docutils<0.20
 ipython_genutils
-importlib_resources
+importlib_resources; python_version < '3.9'

--- a/ci/requirements_rtd.txt
+++ b/ci/requirements_rtd.txt
@@ -11,3 +11,4 @@ nbsphinx
 sphinx-rtd-theme
 docutils<0.20
 ipython_genutils
+importlib_resources

--- a/ci/requirements_travis.txt
+++ b/ci/requirements_travis.txt
@@ -17,3 +17,4 @@ matplotlib
 #pyopencl
 numexpr != 2.8.6
 silx >= 2
+importlib_resources

--- a/ci/requirements_travis.txt
+++ b/ci/requirements_travis.txt
@@ -17,4 +17,4 @@ matplotlib
 #pyopencl
 numexpr != 2.8.6
 silx >= 2
-importlib_resources
+importlib_resources; python_version < '3.9'

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,8 @@ dependencies = [
    'silx>=2',
    'numexpr != 2.8.6',
    'scipy',
-   'matplotlib'
+   'matplotlib',
+   "importlib_resources; python_version < '3.9'"
 ]
 
 [build-system]

--- a/requirements.txt
+++ b/requirements.txt
@@ -13,3 +13,4 @@ PyOpenGL
 PyQt5
 hdf5plugin
 nbsphinx
+importlib_resources

--- a/requirements.txt
+++ b/requirements.txt
@@ -13,4 +13,4 @@ PyOpenGL
 PyQt5
 hdf5plugin
 nbsphinx
-importlib_resources
+importlib_resources; python_version < '3.9'

--- a/src/pyFAI/resources/__init__.py
+++ b/src/pyFAI/resources/__init__.py
@@ -1,7 +1,7 @@
 # coding: utf-8
 # /*##########################################################################
 #
-# Copyright (C) 2016-2018 European Synchrotron Radiation Facility
+# Copyright (C) 2016-2024 European Synchrotron Radiation Facility
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal
@@ -56,7 +56,7 @@ of this modules to ensure access across different distribution schemes:
 
 __authors__ = ["V.A. Sole", "Thomas Vincent"]
 __license__ = "MIT"
-__date__ = "17/05/2019"
+__date__ = "06/03/2024"
 
 
 import os
@@ -65,14 +65,20 @@ import logging
 
 logger = logging.getLogger(__name__)
 
-# pkg_resources is useful when this package is stored in a zip
-# When pkg_resources is not available, the resources dir defaults to the
+# importlib_resources is useful when this package is stored in a zip
+# When importlib.resources is not available, the resources dir defaults to the
 # directory containing this module.
 try:
-    import pkg_resources
+    import  importlib_resources
 except ImportError:
+    logger.info("Unable to import importlib_resources")
     logger.debug("Backtrace", exc_info=True)
-    pkg_resources = None
+    importlib_resources = None
+else:
+    import atexit
+    from contextlib import ExitStack
+    file_manager = ExitStack()
+    atexit.register(file_manager.close)
 
 
 # For packaging purpose, patch this variable to use an alternative directory
@@ -115,11 +121,15 @@ def resource_filename(resource):
 
     if _RESOURCES_DIR is not None:  # if set, use this directory
         return os.path.join(_RESOURCES_DIR, *resource.split('/'))
-    elif pkg_resources is None:  # Fallback if pkg_resources is not available
+    elif importlib_resources is None:  # Fallback if pkg_resources is not available
         return os.path.join(os.path.abspath(os.path.dirname(__file__)),
                             *resource.split('/'))
     else:  # Preferred way to get resources as it supports zipfile package
-        return pkg_resources.resource_filename(__name__, resource)
+        ref = importlib_resources.files(__name__) / resource.dat
+
+        path = file_manager.enter_context(importlib_resources.as_file(ref))
+        print(path)
+        return path
 
 
 _integrated = False

--- a/src/pyFAI/resources/__init__.py
+++ b/src/pyFAI/resources/__init__.py
@@ -56,7 +56,7 @@ of this modules to ensure access across different distribution schemes:
 
 __authors__ = ["V.A. Sole", "Thomas Vincent"]
 __license__ = "MIT"
-__date__ = "06/03/2024"
+__date__ = "07/03/2024"
 
 
 import os
@@ -68,17 +68,21 @@ logger = logging.getLogger(__name__)
 # importlib_resources is useful when this package is stored in a zip
 # When importlib.resources is not available, the resources dir defaults to the
 # directory containing this module.
-try:
-    import  importlib_resources
-except ImportError:
-    logger.info("Unable to import importlib_resources")
-    logger.debug("Backtrace", exc_info=True)
-    importlib_resources = None
+if sys.version_info > (3,9):
+    import importlib.resources as importlib_resources
 else:
-    import atexit
-    from contextlib import ExitStack
-    file_manager = ExitStack()
-    atexit.register(file_manager.close)
+    try:
+        import  importlib_resources
+    except ImportError:
+        logger.info("Unable to import importlib_resources")
+        logger.debug("Backtrace", exc_info=True)
+        importlib_resources = None
+
+if importlib_resources is not None:
+        import atexit
+        from contextlib import ExitStack
+        file_manager = ExitStack()
+        atexit.register(file_manager.close)
 
 
 # For packaging purpose, patch this variable to use an alternative directory
@@ -125,11 +129,9 @@ def resource_filename(resource):
         return os.path.join(os.path.abspath(os.path.dirname(__file__)),
                             *resource.split('/'))
     else:  # Preferred way to get resources as it supports zipfile package
-        ref = importlib_resources.files(__name__) / resource.dat
-
+        ref = importlib_resources.files(__name__) / resource
         path = file_manager.enter_context(importlib_resources.as_file(ref))
-        print(path)
-        return path
+        return str(path)
 
 
 _integrated = False

--- a/src/pyFAI/resources/__init__.py
+++ b/src/pyFAI/resources/__init__.py
@@ -68,7 +68,7 @@ logger = logging.getLogger(__name__)
 # importlib_resources is useful when this package is stored in a zip
 # When importlib.resources is not available, the resources dir defaults to the
 # directory containing this module.
-if sys.version_info > (3,9):
+if sys.version_info >= (3,9):
     import importlib.resources as importlib_resources
 else:
     try:

